tests/e2e/mgmt_wrkld_domain_isolation.go

```diff
@@ -841,4 +875,316 @@ var _ bool = ginkgo.Describe("[domain-isolation] Management-Workload-Domain-Isol
+		storageProfileIdZ2 := e2eVSphere.GetSpbmPolicyID(storagePolicyNameZ2)
+
+		// append late-binding now as it knowns to k8s and not to vc
+		storagePolicyNameZ2 = storagePolicyNameZ2 + "-latebinding"
+
+		// read datastore url
+		zonal2DsUrl := os.Getenv(envZonal2DatastoreUrl)
+
+		// fetch shared vsphere datatsore url
+		datastoreURL := GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		dsRef := getDsMoRefFromURL(ctx, datastoreURL)
+		framework.Logf("dsmoId: %v", dsRef.Value)
+
+		// read or create content library if it is empty
+		if contentLibId == "" {
+			contentLibId, err = createAndOrGetContentlibId4Url(vcRestSessionId, GetAndExpectStringEnvVar(envContentLibraryUrl),
+				dsRef.Value)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Create a WCP namespace tagged to zone-1 & zone-2")
+		namespace, statuscode, err = createtWcpNsWithZonesAndPolicies(vcRestSessionId,
+			[]string{sharedStorageProfileId, storageProfileIdZ1, storageProfileIdZ2}, getSvcId(vcRestSessionId),
+			[]string{zone1, zone2}, vmClass, contentLibId)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(statuscode).To(gomega.Equal(status_code_success))
+		defer func() {
+			delTestWcpNs(vcRestSessionId, namespace)
+			gomega.Expect(waitForNamespaceToGetDeleted(ctx, client, namespace, poll, pollTimeout)).To(gomega.Succeed())
+		}()
+
+		ginkgo.By("Fetch storage class tagged to wcp namespace")
+		storageClassNames := []string{sharedStoragePolicyName, storagePolicyNameZ1, storagePolicyNameZ2}
+		storageClasses := make([]*storagev1.StorageClass, len(storageClassNames))
+		for i, name := range storageClassNames {
+			sc, err := client.StorageV1().StorageClasses().Get(ctx, name, metav1.GetOptions{})
+			if !apierrors.IsNotFound(err) {
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+			storageClasses[i] = sc
+		}
+		sharedStorageclass := storageClasses[0]
+		zonal1Sc := storageClasses[1]
+		zonal2Sc := storageClasses[2]
+
+		ginkgo.By("Create 3 PVCs with different binding modes")
+		storageClassList := []*storagev1.StorageClass{sharedStorageclass, zonal1Sc, zonal2Sc}
+		pvcList := []*v1.PersistentVolumeClaim{}
+		for _, sc := range storageClassList {
+			pvc, err := createPVC(ctx, client, namespace, labelsMap, "", sc, "")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pvcList = append(pvcList, pvc)
+		}
+
+		pvclaim1 := pvcList[0]
+		pvclaim2 := pvcList[1]
+		pvclaim3 := pvcList[2]
+
+		ginkgo.By("Creating a Deployment using pvc-1")
+		dep, err := createDeployment(ctx, client, 1, labelsMap, nil, namespace,
+			[]*v1.PersistentVolumeClaim{pvclaim1}, execRWXCommandPod1, false, busyBoxImageOnGcr)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		_, err = fdep.GetPodsForDeployment(ctx, client, dep)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create Pod to attach to Pvc-2")
+		pod2, err := createPod(ctx, client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim2}, false,
```

Comment:
After Pod creation, check volume node attachment
reference code below

		ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s", volumeID2, nodeName2))
		isDiskAttached2, err := e2eVSphere.isVolumeAttachedToVM(client, volumeID2, vmUUID2)
		gomega.Expect(err).NotTo(gomega.HaveOccurred())
		gomega.Expect(isDiskAttached2).To(gomega.BeTrue(), "Volume is not attached to the node")